reactor-aeron-core/src/main/java/reactor/aeron/AeronClientConnector.java

```diff
@@ -124,21 +137,25 @@
             });
   }
 
-  private Mono<MessagePublication> tryConnect() {
+  private Mono<MessagePublication> tryConnect(AeronEventLoop eventLoop) {
     return Mono.defer(
         () -> {
           int retryCount = options.connectRetryCount();
           Duration retryInterval = options.connectTimeout();
 
           // outbound->Pub(endpoint, sessionId)
-          return Mono.defer(() -> resources.publication(getOutboundChannel(), STREAM_ID, options))
+          return Mono.defer(() -> getPublication(eventLoop))
               .flatMap(mp -> mp.ensureConnected().doOnError(ex -> mp.dispose()))
               .retryBackoff(retryCount, Duration.ZERO, retryInterval)
               .doOnError(
                   ex -> logger.warn("aeron.Publication is not connected after several retries"));
         });
   }
 
+  private Mono<MessagePublication> getPublication(AeronEventLoop eventLoop) {
+    return resources.publication(getOutboundChannel(), STREAM_ID, options, eventLoop);
+  }
+
```

Comment:
I know this method was added because of checkstyle;  but   now it looks  like  inlining "getOutboundChannel"  would make very much sense, IMHO